### PR TITLE
Update UPE settings block for BNLP

### LINKED ISSUE #6276: [BNPL][MVP] Update UPE Notification and Enablement Card on Payments Settings Screen
### Description

**As a WooPayments merchant eligible for UPE,**
_When I visit the Payments > Settings screen and I don't have any UPE method enabled,_
_I want to see a card that informs me about the UPE offerings and contains a button to take me to the enable UPE screen._

### Acceptance criteria

1. Display a card on the Payments > Settings screen for eligible WooPayments merchants who don't have any UPE methods enabled.
2. The card should inform merchants about the UPE offerings.
3. The card should contain a button that takes merchants to the enable UPE screen.

### Designs

Screenshot 2023-05-11 at 21 05 36

### Dev notes

* We do not display the card if they have at least one UPE method enabled
* The functionality of displaying the card is already there, this is purely design change
* [Figma prototype](W6rxUkOIo65PmoR5idegtQ-fi-1731_105048)

### PR DESCRIPTION
**Pull request by KarlisJ**
Fixes #6276 

Update the UPE settings card to reflect Affirm and Afterpay as payment methods for those merchant who have the feature available.

The new design is tailored to also account for non-eligible merchants by only removing the icons from the asset.

The new design on desktop
image

The new design on Mobile
image

#### Testing instructions

Hardcode the return value of `WC_Payments_Features::is_bnpl_affirm_afterpay_enabled()` method to both `true` and `false` and observe:
- if feature is not enabled, we should see the asset without "Affirm" and "Afterpay" icons
- if feature is enabled, we should see the asset with "Affirm" and "Afterpay" icons.

-------------------

- [ x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by KarlisJ** [comment]
I made a mistake and based this PR off and branch that's not yet in `develop`. Please review only last 2 commits. 

The merge conflicts should be resolved as soon as [Add feature flag for BNLP Affirm Afterpay payment method](https://github.com/Automattic/woocommerce-payments/pull/6288#top) is merged

**Comment by nagpai** [comment]
Manual testing done and it works perfect ✅ 

With BNPL disabled 
image


With BNPL enabled
image

File: changelog/add-6034-feature-flag-bnlp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: dev
+Comment: This merge is a feature flog utilities that will be made available for masses only in later stages of the project
```
Comment (nagpai) [inline changelog/add-6034-feature-flag-bnlp]: Typo in `flag` .

Comment (nagpai) [reply changelog/add-6034-feature-flag-bnlp]: Is this a changelog of a different PR?

**Review by nagpai — APPROVED** [review]
Minor comment on changelog. Preapproving subject to resolving conflicts.

File: changelog/add-6034-feature-flag-bnlp (hunk 1 above)
Comment (KarlisJ) [reply changelog/add-6034-feature-flag-bnlp]: Yes, there's a glitch in GH. It's showing diff to files that are already in `develop`.

File: client/payment-methods/index.js [hunk 2]
```diff
@@ -77,9 +77,41 @@ const UpeSetupBanner = () => {
 		} );
 	};
 
-	return (
-		<>
-			<CardDivider />
+	let card;
```
Comment (mgascam) [inline client/payment-methods/index.js]: I'd suggest a comment explaining that we will remove the condition after the project rollout.

File: client/payment-methods/index.js
Comment (mgascam) [inline client/payment-methods/index.js:1]: Noting that we usually migrate to Typescript when we modify a js file.

Comment (mgascam) [inline client/payment-methods/index.js:1]: Do you think we should add new tests for this file in `client/payment-methods/test/index.js`?

File: client/payment-methods/index.js (hunk 2 above)
Comment (KarlisJ) [reply client/payment-methods/index.js]: Not relavant anymore after [this commit ](https://github.com/Automattic/woocommerce-payments/pull/6386/commits/36d24bfe031810fdd904a2fdea28c48004d61fad)

File: client/payment-methods/index.js
Comment (KarlisJ) [reply client/payment-methods/index.js:1]: I plan to that, right now focusing on the implementation details

Comment (KarlisJ) [reply client/payment-methods/index.js:1]: Yes. Currently we don't have any tests for this component but I'd prefer if we at least created a test for showing correct asset based on feature flag.

That way, once the feature flag will be removed, the test will fail and we'll have to update it to reflect the conditional render on the eligibility of merchant.

**Comment by KarlisJ** [comment]
@nagpai given [this commit](https://github.com/Automattic/woocommerce-payments/pull/6386/commits/36d24bfe031810fdd904a2fdea28c48004d61fad) I've updated testing instructions as we no longer support the old design.

File: client/payment-methods/style.scss [hunk 3]
```diff
@@ -60,6 +60,40 @@
...
+		}
+
+		&-get-started {
+			margin-right: 16px;
+		}
+		&-actions {
+			padding-top: 11px;
+		}
+
+		@media ( max-width: 1024px ) {
```
Comment (mgascam) [inline client/payment-methods/style.scss]: I noticed that the illustration is not hidden in narrow screens when the `.background-local-payment-methods` is active:

Screenshot 2023-06-05 at 11 30 03

Can you have a look please?

**Review by mgascam — CHANGES_REQUESTED** [review]
Finished a manual testing round. I only found a small CSS issue: the illustration overlaps the text in narrow screens when the non-BNPL illustration is displayed. It hides it as expected in BNPL scenario.

Other than that, code looks great. I like the way you manage to reuse the illustration for BNPL and non-BNPL scenarios.

File: client/payment-methods/style.scss (hunk 3 above)
Comment (KarlisJ) [reply client/payment-methods/style.scss]: Good catch, thank you!

Comment (KarlisJ) [reply client/payment-methods/style.scss]: Addressed [here](https://github.com/Automattic/woocommerce-payments/pull/6386/commits/76639cff1128d48c4316f7ad67c3fd7458df97b0).

Ready for final review.

**Review by mgascam — APPROVED** [review]
LGTM!

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6386` or branch name `update/6276-upe-settings-for-bnlp` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=update/6276-upe-settings-for-bnlp) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 500a3ab13ed2f66c932ef3133356eeb49e6c5482
- Build time: 2023-06-08 05:30:58 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by nagpai** [comment]
Tested with new instructions 

BNPL available:
image

BNP not available - banner seems to appear with zero padding 

image

Seems to be happening because of this code 

https://github.com/Automattic/woocommerce-payments/blob/15940179275ea19019805a004941bb8538fe8e71/client/payment-methods/style.scss#L5-L6

Something along the lines of this rule will help 

```css
.components-card__body.components-card-body.wcpay-card-body.payment-methods__upe-methods.background-local-payment-methods{
		padding: 24px;
	}
```

**Comment by KarlisJ** [comment]
Thanks @nagpai for checking this. I can reproduce the issue but it's not related to BNPL eligibility. The design broke after merging latest changes in the `develop` branch. 

I'll try to compare what's different and from that find the exact cause.

**Comment by nagpai** [comment]
It looks like this line that got merged in this [conflict resolve commit](https://github.com/Automattic/woocommerce-payments/commit/44d4147bde0e91de36deedd0559108953c37359f). 

https://github.com/Automattic/woocommerce-payments/blame/44d4147bde0e91de36deedd0559108953c37359f/client/payment-methods/style.scss#L4

**Comment by KarlisJ** [comment]
@nagpai @mgascam Latest commit addressed the issue. Ready for final review.

**Review by mgascam — APPROVED** [review]
After latest change, the illustration shows with the correct padding

File: client/payment-methods/index.js [hunk 4]
```diff
@@ -97,7 +101,7 @@ const UpeSetupBanner = () => {
 
 				<div className="payment-methods__express-checkouts-actions">
 					<span className="payment-methods__express-checkouts-get-started">
-						<Button isPrimary onClick={ handleEnableUpeClick }>
+						<Button isSecondary onClick={ handleEnableUpeClick }>
```
Comment (jbordonado) [inline client/payment-methods/index.js:104]: `isPrimary` and `isSecondary` are both deprecated attributes from the `<Button>` component

Next time, could you try to migrate to the new suggested way (`variant="secondary"` in this case) when you encounter deprecated usages, please?

**Pull request merged**